filebeat/harvester/log.go

```diff
@@ -44,6 +44,29 @@ func (h *Harvester) Harvest() {
 		return
 	}
 
+	go func() {
+		select {
+		case <-h.prospectorDone:
+			close(h.done)
+		case <-h.done:
+			return
+		}
+
+	}()
+
+	if h.config.CloseTimeout > 0 {
+		go func() {
+			select {
+			case <-h.done:
+				return
+			case <-time.After(h.config.CloseTimeout):
+				close(h.done)
```

Comment:
Suggestion: I will document this potential issue in the docs for the moment. TBH not 100% sure if that is what would happen. I think it is more that the processor will block the timeout as even though the channel is closed, I wont have an affect on the processor currently. It would be nice to enhance the processors in a way that it is possible to stop a processor.